smarts/core/sumo_traffic_simulation.py

```diff
@@ -692,13 +748,15 @@ def _reroute_vehicles(self, vehicle_states):
             new_route_edges = route_edges[-1:] + route_edges
             self._traci_conn.vehicle.setRoute(vehicle_id, new_route_edges)
 
```

Comment:
Should the error handling about traci connection be done here and for any reference call to `self._traci_conn` in this file.
```python
if not self.connected:
            return []
        try:
            self._traci_conn.vehicle.setRoute(vehicle_id, new_route_edges)
        except self._traci_exceptions as e:
````

Reply:
I wrapped these calls at a higher level. I only do exception handling and connection checks at the public method body level to reduce the amount of checks needed.